File: esempi-api/caricafoto.swift
import Foundation

let headers = [
  "content-type": "multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW",
  "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJwcm92YSIsImV4cCI6MTQ5NTUyNjQ0MTkyMX0.NYeq4xAiAo8iqDBK7LGkuE6FNCAXgsJE5Lhkw-w0N6g",
  "cache-control": "no-cache",
  "postman-token": "395879f9-cfb8-11c7-3e60-d67c4312b6cf"
]
let parameters = [
  [
    "name": "foto",
    "fileName": "C:\Users\nicho\Desktop\idee.png"
  ]
]

let boundary = "----WebKitFormBoundary7MA4YWxkTrZu0gW"

var body = ""
var error: NSError? = nil
for param in parameters {
  let paramName = param["name"]!
  body += "--\(boundary)\r\n"
  body += "Content-Disposition:form-data; name=\"\(paramName)\""
  if let filename = param["fileName"] {
    let contentType = param["content-type"]!
    let fileContent = String(contentsOfFile: filename, encoding: String.Encoding.utf8)
    if (error != nil) {
      print(error)
    }
    body += "; filename=\"\(filename)\"\r\n"
    body += "Content-Type: \(contentType)\r\n\r\n"
    body += fileContent
  } else if let paramValue = param["value"] {
    body += "\r\n\r\n\(paramValue)"
  }
}

let request = NSMutableURLRequest(url: NSURL(string: "https://lalapark.it/api/user/pic")! as URL,
                                        cachePolicy: .useProtocolCachePolicy,
                                    timeoutInterval: 10.0)
request.httpMethod = "POST"
request.allHTTPHeaderFields = headers
request.httpBody = postData as Data

let session = URLSession.shared
let dataTask = session.dataTask(with: request as URLRequest, completionHandler: { (data, response, error) -> Void in
  if (error != nil) {
    print(error)
  } else {
    let httpResponse = response as? HTTPURLResponse
    print(httpResponse)
  }
})

dataTask.resume()

File: esempi-api/caricafoto.java
  new Thread(new Runnable() {
            @Override
            public void run() {
                try {
                    OkHttpClient client = new OkHttpClient();

                    MultipartBody.Builder builder = new MultipartBody.Builder()
                            .setType(MultipartBody.FORM);

                    MediaType mediaType = MediaType.parse("image/png");
                    builder.addFormDataPart("foto",fileToUp.getName(),RequestBody.create(mediaType, fileToUp));
                    RequestBody body = builder.build();

                    okhttp3.Request request = new okhttp3.Request.Builder()
                            .url(URL)
                            .post(body)
                            .addHeader("content-type", "multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW")
                            .addHeader("token", userToken)
                            .addHeader("cache-control", "no-cache")
                            .build();


                    okhttp3.Response response = client.newCall(request).execute();



                    if(DEBUG)
                        Log.d(TAG_FRAGMENT,mediaType.toString());

                    if(DEBUG)
                        Log.d(TAG_FRAGMENT,response.body().string());

                } catch (IOException | NullPointerException e) {
                    e.printStackTrace();
                }

            }
        }).start();
